File: src/leetcode/python/SumOfSquareNumbers_633.py
class SumOfSquareNumbers_633(object):
    def judgeSquareSum(self, c):
        """
        :type c: int
        :rtype: bool
        """
        import math
        left, right = 0, int(math.sqrt(c))
        while left <= right:
            res = left * left + right * right
            if res == c:
                return True
            if res < c:
                left += 1
            else:
                right -= 1
        return False

File: src/leetcode/java/SumOfSquareNumbers_633.java
package leetcode.java;

public class SumOfSquareNumbers_633 {
  public boolean judgeSquareSum(int c) {
    int left = 0, right = (int) Math.sqrt(c);
    while (left <= right) {
      int res = left * left + right * right;
      if (res == c) return true;
      if (res < c) left++;
      else right--;
    }
    return false;
  }
}
